### DISCUSSION TITLE
MGCB can't resolve assemblies after upgrade from 3.7.1

### DISCUSSION THREAD

**Discussion by JoccyGitHub** [MonoGame/MonoGame#8204] [Content]
Hello everyone,

Hope this is the right place to ask this question, as I'm a total newbie to GitHub.

I've been trying to upgrade an older MonoGame project of mine, still on 3.7.1 and being developed with VS2019, to the current state-of-the-art.

I installed VS2022 w/ the MonoGame template extension, made sure everything was updated, ran successfully the default project, and then proceeded to reimport all my stuff in a new Desktop Application project by following the guide on the website.

Since my project relies on referencing other projects, I also redid their .csproj files and they all rebuild successfully.

Problem is, **now MGCB cannot seemingly resolve two assemblies needed by a custom font processor** used in my project (this one specifically: https://www.axiomverge.com/blog/localized-font-textures-in-monogame ).

Back in the day, on MonoGame 3.7.1, I compiled this reworked ttf2bmp.exe, I referenced it within my Content.mgcb, and it worked very well.
After upgrading to 3.8.1.303, I get this error while trying to use its importer:

```
Failed to load assembly 'D:/(Project Path)/TTF2BMPP/ttf2bmp.exe': Unable to load one or more of the requested types.
Could not load file or assembly 'System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=xxx'. The system cannot find the file specified.
Could not load file or assembly 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=yyy'. The system cannot find the file specified.
```

Note: didn't paste here their actual PublicKeyTokens, doubt they matter.

Things I am doing:

1. Targeting net6.0 with AnyCPU in all the involved projects
2. Referencing ttf2bmpp with a relative path like this: ` /reference:..\..\TTF2BMPP\ttf2bmp.exe` to allow for relocating my repository

Things I have tried:

1. Reinstall .NET (both 6.0 and 8.0): no effect whatsoever
2. Change the /reference path to ttf2bmpp to absolute in Content.mgcb: no effect whatsoever
3. Recompile ttf2bmpp in the new VS2022 environment with target AnyCPU: rebuilds correctly and boots up when running the exe, but no effect whatsoever
4. Upgrade ttf2bmpp and its related .csproj to net6.0-windows (due to the need for System.Windows.Forms) and recompile it successfully with target AnyCPU: still boots up when running the exe, but I'm getting a different error in MGCB because now the target platform (Windows) is different from my main project (targeted to generic net6.0)
5. Upgrade ttf2bmpp and its related .csproj to net6.0 and link the .dll instead of the .exe: getting the same error as always, except it now lists version 6.0.2.0 for System.Windows.Forms instead of 4.0.0.0 (I guess 6.0.2.0 is the net6.0 version of the same assembly)
6. Reference the missing assemblies in the .csproj of my project:
```
  <ItemGroup>
    <Reference Include="System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=xxx" />
    <Reference Include="System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=yyy" />
  </ItemGroup>
```
...this got them to show in my Solution Explorer, and I did notice they had a warning icon (meaning VS could not resolve them as well? could this be a hint?). If possible I'd rather not have to reference them in my project though.
7. Switch temporarily to standard MonoGame importers for the affected fonts: **in this case the project builds successfully** and seems to be working fine except for the text (which is, of course, all garbled up due to the incorrect importer in use)

**...what am I doing wrong?**

Thank you all in advance!

**Comment by JoccyGitHub** [comment]
I managed to resolve this, by the way.

1. I noticed there was a Nuget package for System.Configuration.ConfigurationManager v6.0.0.0, so I added a reference to it in my version of ttf2bmpp that I had updated for net6.0-windows (see point 5 in "things I have tried" above);
2. For System.Windows.Forms I just excised the Windows app part, in order to build the font processor only as a .dll (without this, I could change the ttf2bmpp project back to generic net6.0). I can keep the full application as a separate project, and only use it whenever necessary.

Not the neatest solution, but I guess it will have to do for the time being.

**Comment by landon-codes** [comment]
I had a similar issue when going through the DungeonSlime tutorial where MonoGame was missing a library to compile .fx files.
I ended up having to skip that chapter.
